torch/_dynamo/backends/distributed.py

```diff
@@ -464,13 +463,30 @@ def __init__(
     def _ignore_parameter(self, parameter):
         return hasattr(parameter, "_ddp_ignored") and parameter._ddp_ignored
 
+    def add_param(self, bucket, param, name):
+        bucket.size += param.untyped_storage().nbytes()
+        bucket.params.append(name)
+        bucket.param_ids.append(id(param))
+
     def add_module_params_to_bucket(self, mod, bucket, processed_modules, prefix):
         processed_modules.add(mod)
         for name, param in mod.named_parameters():
             if param.requires_grad and not self._ignore_parameter(param):
-                bucket.size += param.untyped_storage().nbytes()
-                bucket.params.append(f"{prefix}_{name}")
-                bucket.param_ids.append(id(param))
+                self.add_param(bucket, param, f"{prefix}.{name}")
```

Comment:
can you motivate the change from `prefix_name` to `prefix.name` and argue why it's always a safe change?

i assume part of the motivation is that the function would refer to a parameter via `prefix.name`, but i'm not sure if the safety part has been thought through fully.

Reply:
Yes, I agree. @laithsakka any reason you preferred `_` over `.`?